### PR TITLE
Expand AnonymousRequestMatcher by user_identifier_headers option

### PR DESCRIPTION
**Pull request by dbu**
continuation of #408

### CODE REVIEW

File: src/UserContext/AnonymousRequestMatcher.php [hunk 1]
```diff
@@ -13,36 +13,53 @@
...
-            return false;
+        foreach ($this->options['user_identifier_headers'] as $header) {
+            if ($request->headers->has($header)) {
+                return false;
+            }
         }
 
         foreach ($request->cookies as $name => $value) {
-            if (0 === strpos($name, $this->sessionNamePrefix)) {
+            if (0 === strpos($name, $this->options['session_name_prefix'])) {
```
Comment (dbu) [inline src/UserContext/AnonymousRequestMatcher.php]: @blankse wdyt, would it make sense to make it possible to ignore cookies? the old code the bundle had did make it possible by not having the Cookie header in the list of headers... could we say that if the prefix is `''` we do not check cookies? or allow type boolean as well and set the prefix to false?

Comment (dbu) [reply src/UserContext/AnonymousRequestMatcher.php]: i pushed a change to allow that. @blankse can you take a look at e60ad47a4dfe6db929a395bd6c98af9f868c2a2e and tell me if you like it or not?

Comment (blankse) [reply src/UserContext/AnonymousRequestMatcher.php]: @dbu LGTM :+1:

**Comment by dbu** [comment]
i tagged 1.4.5 with this and will now merge the changes up to the master branch so they will be in 2.2 when that is released.

**Pull request merged**